app/scripts/views/settings/clients.js

```diff
@@ -135,11 +141,20 @@ define(function (require, exports, module) {
       // if a device then ask for confirmation
       if (clientType === Constants.CLIENT_TYPE_DEVICE) {
         this.navigate('settings/clients/disconnect', {
-          clientId: client.get('id'),
+          clientId: client.get('deviceId'),
           clients: this._attachedClients
         });
       } else {
-        this.user.destroyAccountClient(this.user.getSignedInAccount(), client);
+        this.user.destroyAccountClient(this.user.getSignedInAccount(), client)
+          .then(() => {
+            if (clientType === Constants.CLIENT_TYPE_WEB_SESSION) {
+              return this.user.sessionStatus().then((res) => {
+              }, (err) => {
+                // if disconnected the current session, the user is signed out
+                this.navigateToSignIn();
```

Comment:
Could you add the [signed-out-mixin](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/views/mixins/signed-out-notification-mixin.js) and have this done automatically?